src/hooks/useVerifiableCredential.test.ts

```diff
@@ -0,0 +1,247 @@
+    });
+
+    mockUseToast.mockReturnValue(toastSpy);
+
+    return {
+      signTypedDataAsyncSpy,
+      verifyTypedDataSpy,
+      toastSpy,
+      mockGetAddress,
+      mockUseAccount,
+    };
+  };
+
+  it('returns correct properties', async () => {
+    const { mockUseAccount } = buildMock();
+    mockUseAccount.mockReturnValue({ address: VALID_ACCOUNT_1 });
+
+    const { result } = await act(() =>
+      renderHook(() => useVerifiableCredential()),
+    );
+
+    expect(result.current.issuerAddress).toBe(VALID_ACCOUNT_1);
+    expect(
+      result.current.accountVCBuilder instanceof AccountVerifiableCredential,
+    ).toBe(true);
+    expect(
+      result.current.snapVCBuilder instanceof SnapVerifiableCredential,
+    ).toBe(true);
+    expect(typeof result.current.signMessage === 'function').toBe(true);
+    expect(result.current.signatureError).toBeUndefined();
+  });
+
+  describe('signMessage', () => {
+    it('signs and verify message if address is not empty', async () => {
+      const { mockUseAccount, signTypedDataAsyncSpy, verifyTypedDataSpy } =
+        buildMock();
+      mockUseAccount.mockReturnValue({ address: VALID_ACCOUNT_1 });
+      signTypedDataAsyncSpy.mockResolvedValue('0xSignature');
+      verifyTypedDataSpy.mockResolvedValue(true);
+
+      const { result } = await act(() =>
+        renderHook(() => useVerifiableCredential()),
+      );
+
+      const vc = result.current.accountVCBuilder.buildAccountTrust(
+        VALID_ACCOUNT_1,
+        VALID_ACCOUNT_2,
+      );
+      const signResult = await act(async () => result.current.signMessage(vc));
+
+      expect(signTypedDataAsyncSpy).toHaveBeenCalled();
+      expect(verifyTypedDataSpy).toHaveBeenCalled();
+      expect(signResult).toBe('0xSignature');
+    });
+
+    it('does not sign and verify message if address is empty', async () => {
+      const { mockUseAccount, verifyTypedDataSpy, signTypedDataAsyncSpy } =
+        buildMock();
+      mockUseAccount.mockReturnValue({ address: null });
+
+      const { result } = await act(() =>
+        renderHook(() => useVerifiableCredential()),
+      );
+
+      const vc = result.current.accountVCBuilder.buildAccountTrust(
+        VALID_ACCOUNT_1,
+        VALID_ACCOUNT_2,
+      );
+      const signResult = await act(async () => result.current.signMessage(vc));
+
+      expect(signTypedDataAsyncSpy).not.toHaveBeenCalled();
+      expect(verifyTypedDataSpy).not.toHaveBeenCalled();
+      expect(signResult).toBeNull();
+    });
+
+    it('does not trigger verify if signature is null', async () => {
+      const { mockUseAccount, verifyTypedDataSpy, signTypedDataAsyncSpy } =
+        buildMock();
+      mockUseAccount.mockReturnValue({ address: VALID_ACCOUNT_1 });
+      signTypedDataAsyncSpy.mockResolvedValue(null);
+
+      const { result } = await act(() =>
+        renderHook(() => useVerifiableCredential()),
+      );
+
+      const vc = result.current.accountVCBuilder.buildAccountTrust(
```

Comment:
minor : common in all tests. can be reused

Reply:
yup, will update it